front/src/common/authorization/hooks/useUserResourcesGrants.ts

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState } from 'react';
+
+import { assign } from 'lodash';
+
+import {
+  mockedEditoastApi,
+  type GetGrantsByResourceTypeResponse,
+  type ResourceType,
+} from 'common/api/mock/mockEditoastApi';
+
+type ResourcesGrants = Partial<Record<ResourceType, GetGrantsByResourceTypeResponse>>;
+
+/**
+ * provides the user grants on a list of resources
+ * @param payload the resource type and the resource ids
+ * @returns the user grants on the resources
+ */
+const useUserResourcesGrants = (payload: Partial<Record<ResourceType, number[]>>) => {
+  const [resourcesGrants, setResourcesGrants] = useState<ResourcesGrants>({});
+  const [fetchUserGrantByResourceId, { data: userResourcesGrants }] =
+    mockedEditoastApi.endpoints.postUserResourcesGrants.useMutation();
+  const [getGrantsByResourceType] =
+    mockedEditoastApi.endpoints.getGrantsByResourceType.useLazyQuery({});
+
+  const resourcesTypes = Object.keys(payload) as ResourceType[];
+
+  useEffect(() => {
+    const getUserGrantByResourceId = async () => {
+      try {
+        const validPayload = Object.entries(payload).reduce<
+          Partial<Record<ResourceType, number[]>>
+        >((acc, [resourceType, ids]) => {
+          if (ids.length > 0) {
+            acc[resourceType as ResourceType] = ids;
+          }
+          return acc;
+        }, {});
+
+        if (Object.keys(validPayload).length > 0) {
+          await fetchUserGrantByResourceId(validPayload);
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    const fetchGrantsByResourceType = async () => {
```

Comment:
nit:
```suggestion
    const fetchGrantTreeByResourceType = async () => {
      const newGrantTreeByResourceType = new Map<ResourceType, PrivilegesByGrant>();
      await Promise.all(
        Array.from(payload.keys()).map(async (resourceType) => {
          const { data } = await getGrantsByResourceType(resourceType);
          newGrantTreeByResourceType.set(resourceType, data!);
        })
      );
      setGrantTreeByResourceType(newGrantTreeByResourceType);
    };
```

`newGrantTreeByResourceType` is a map. It is updated as soon as a promise is resolved